src/Maestro/maestro-angular/src/app/model/version-details.ts

```diff
@@ -0,0 +1,362 @@
+
+export class VersionDetails {
+
+  // <DependencyName, DependencyDetails>
+  public allDependencies: Record<string, DependencyDetail> = {};
+  public dependenciesForEvaluation: Record<string, DependencyDetail> = {};
+
+  // <SubId, Subscription>
+  public unusedSubscriptions: Record<string, Subscription> = {};
+  public conflictingSubscriptions: Record<string, Subscription[]> = {};
+  public unableToRetrieveAssetsFor: Record<string, Subscription> = {};
+
+  // Dependency Name
+  public dependenciesWithNoSubscription: string[] = new Array();
+
+  // Subscription ID
+  public isToolsetSubscription: Record<string, boolean> = {};
+
+  constructor(inputFile: XMLDocument) {
+    const productElements = inputFile.getElementsByTagName("ProductDependencies");
+
+    if (productElements.length != 0) {
+      let childProductElements = productElements[0].getElementsByTagName('Dependency');
+
+      for (let el of Array.from(childProductElements)) {
+        this.parseDependencyElement(el, this.allDependencies, false);
+      }
+    }
+
+    const toolsetElements = inputFile.getElementsByTagName("ToolsetDependencies");
+
+    if (toolsetElements.length != 0) {
+      let childToolsetElements = toolsetElements[0].getElementsByTagName('Dependency');
+
+      for (let el of Array.from(childToolsetElements)) {
+        this.parseDependencyElement(el, this.allDependencies, true);
+      }
+    }
+
+    const depNames = Object.keys(this.allDependencies);
+    for (let dep of depNames) {
+      if (!this.allDependencies[dep].pinned && !this.allDependencies[dep].coherentParentDependencyName) {
+        this.dependenciesForEvaluation[dep] = this.allDependencies[dep];
+      }
+    }
+  }
+
+  parseDependencyElement(element: Element, dependencyList: Record<string, DependencyDetail>, toolset: boolean) {
+    let details = new DependencyDetail();
+
+    details.fromToolset = toolset;
+
+    const attributesForElement = element.attributes;
+    const nameAttribute = attributesForElement.getNamedItem("Name");
+
+    if (nameAttribute) {
+      details.name = nameAttribute.nodeValue || "UnknownAssetName";
+    }
+    else {
+      details.name = "UnknownAssetName";
+    }
+
+    const uriAttribute = element.getElementsByTagName("Uri").item(0);
+
+    if (uriAttribute) {
+      details.repoUrl = uriAttribute.textContent || "UnknownUri";
+    }
+    else {
+      details.repoUrl = "UnknownUri";
+    }
+
+    const coherentParent = attributesForElement.getNamedItem("CoherentParentDependency");
+
+    if (coherentParent) {
+      details.coherentParentDependencyName = coherentParent.textContent || undefined;
+    }
+
+    const pinned = attributesForElement.getNamedItem("Pinned");
+
+    if (pinned) {
+      details.pinned = pinned.textContent === 'true';
+    }
+
+    dependencyList[details.name] = details;
+  }
+
+  checkIfToolsetSubscriptions(subsAndAssets: Record<string, Asset[]>) {
+    // A build is a "toolset" if it has at least one dependency pointing to it,
```

Comment:
Can you clarify this comment?

Reply:
I copied the definition of a toolset from the build-graph-table component isToolset code since the subscription objects don't have a concept of toolset or product.